### ISSUE TITLE
Removal of delay breaks set generator

### ISSUE THREAD

**Issue by mjmckp** [fscheck/FsCheck#131]
Prior to version 2.0.5, the following generator worked fine to generate a set of size `n` from a generator `g`:

```
    let genSetOfSize n g =
        gen { let s = ref Set.empty
              while s.Value.Count < n do
                let! value = g
                s := s.Value.Add(value)
              return s.Value
            }
```

Now this always returns the empty set.

(This could be written recursively, but I've had issues with stack overflow in Debug build with this approach)

**Comment by kurtschelfthout** [comment]
Thanks. I got a bit cheeky and removed the delay on while / for as well. Clearly a mistake. I'll put it back, but will have to wait until the weekend as I'm out this week.

**Comment by kurtschelfthout** [comment]
This should be fixed; I have reverted those changes and released 2.0.6.

**Comment by mjmckp** [comment]
Thanks Kurt
